### DISCUSSION TITLE
Does serverless v4 work with Typescript + ECR?

### DISCUSSION THREAD

**Discussion by edwmurph** [serverless/serverless#12731] [General]
Is there a good way with Serverless v4 to write a lambda in Typescript that is configured as an ECR image rather than uploading a zip? Also for my use case I would like to be able to use serverless-offline for local development

It looks like the native [esbuild serverless v4 feature won't compile the typescript if the lambda is configured as an ECR image](https://github.com/serverless/serverless/blob/fb8a49dabf3e9062dee6540e1e995946c3cbd4c0/lib/plugins/esbuild/index.js#L209-L212)?

One change that would help is if there is a command I can use to manually compile via esbuild the same way that would happen if the lambda wasn't using an ECR image. `sls package` doesn't compile the lambda function TS if the lambda is using an ECR image

**Comment by rbro112** [comment]
Seconding this as we'd love this functionality. We have a typescript function we'd like to leverage the default serverless esbuild for rather than building/packaging manually in a custom image. 

Our use case is we essentially just want to provide some additional dependencies in our custom image built on top of the standard NodeJS runtime, and have serverless compile the handler on our behalf. We could then have our dockerfile pull the compiled code, making our life far easier, as manually wrangling with esbuild has been essentially impossible to get working.
